content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "From code search toward a code intelligence platform"
+publishDate: 2022-09-27T10:00-07:00
+description: Finding things in code is good, but learning things about code is great. Sourcegraph's code intelligence platform is built to help.
+tags: [blog]
+slug: "code-search-to-code-intelligence"
+published: false
+heroImage: 
+socialImage: 
+---
+
+No matter how good your other dev tools, coding is still complex. Even if you've configured your editor just right, sped up your test suite, set up continuous deployment, added observability, etc., coding is still complex. Higher-level languages, libraries, frameworks, cloud providers, etc., that encapsulate complexity just let you spend your complexity budget elsewhere. The overall complexity of code has increased, not declined, over time, and there's no sign of that trend changing.
```

Comment:
We say "Coding is still complex" 2x, and then "The overall complexity of code has increased" — the repetition hits the point home, but I don't think that's the way we want to take it.

Reply:
@amieroth Do you think the literal repetition ("coding is complex") doesn't read well, or do you not like complexity being the narrative focus for the intro?

Reply:
```suggestion
No matter how good your dev tools are, coding is complex. This holds true even if you've configured your editor just right, sped up your test suite, set up continuous deployment, and added observability. Tools like higher-level languages, libraries, and frameworks can help, but as you take on bigger and more ambitious projects, you'll likely spend your complexity budget elsewhere. The overall complexity of code has increased over time, and there's no sign of that trend changing.
```

If the former, here is a suggestion with a little less repetition -

Reply:
No matter how good your dev tools are, coding is complex. This is true even if you've configured your editor just right, sped up your test suite, set up continuous deployment, and added observability. Tools like higher-level languages, libraries, and frameworks can help, but as you take on bigger and more ambitious projects, you'll likely spend your complexity budget elsewhere. The overall complexity of code has increased over time, and there's no sign of that trend changing.

Reply:
```suggestion
No matter how good your dev tools are, coding is complex. This is true even if you've configured your editor just right, sped up your test suite, set up continuous deployment, and added observability. Tools like higher-level languages, libraries, and frameworks can help, but as you take on bigger and more ambitious projects, you'll likely spend your complexity budget elsewhere. The overall complexity of code has increased over time, and there's no sign of that trend changing.
```